DA-215: Allow jumps to annotations that have not been rendered

stgrue opened the pull request:
Previously, selecting an annotation that had not been rendered before
would not result in a jump towards that annotation. This has been fixed.
Furthermore, some of the code in d3Annotation.js has been restructured
to make it more readable. For example, all variable declarations have
been moved to the top of the file and whitespace has been added between
function definitions.

JeannedArk commented on src/main/webapp/directives/d3Annotation.js:
```diff
@@ -57,8 +52,7 @@ angular
                 const wordSpacing = 3.5;
```
Can you add a comment to each constant? "oma" and "opa" are still not understandable.

JeannedArk commented on src/main/webapp/directives/d3Annotation.js:
```diff
@@ -1851,14 +1892,14 @@ angular
                     if (minLine !== undefined && maxLine !== undefined) {
                         const min = formAnnotation.startLine();
                         const max = formAnnotation.endLine();
                         if (max < minLine || min > maxLine)
```
Transform it to a single return statement, no need for an if-statement

JeannedArk commented on src/main/webapp/directives/d3Annotation.js:
```diff
@@ -40,25 +40,25 @@ angular
-                const wordHeight = 40;
+            link: function ($scope, iElement) {
+                const lineHeight = 80; // Height of each displayed line of text
+                const wordHeight = 80; // Height and position of text in annotation boxes
```
Why position?

JeannedArk commented on src/main/webapp/directives/d3Annotation.js:
```diff
@@ -40,25 +40,25 @@ angular
                 const margin = 80; // Space on top of text
-                const annotationHeight = wordHeight / 2.4;
-                const scale = 1.2;
+                const annotationHeight = wordHeight / 2.4; //
```
Missing comment

JeannedArk commented on src/main/webapp/directives/d3Annotation.js:
```diff
@@ -5,14 +5,14 @@
  */
```
There are a lot of "Rounded values increase performance" comments. You can remove them and insert this comment instead one time in the header of this file.

Commit:
DA-215: Allow jumps to annotations that have not been rendered

Previously, selecting an annotation that had not been rendered before
would not result in a jump towards that annotation. This has been fixed.
Furthermore, some of the code in d3Annotation.js has been restructured
to make it more readable.
